alphagov/paas-cf
# [#109030954]  Concourse job to create minimal VPC

# What

Two concourse pipelines to set up initial VPC + S3 bucket and to destroy them on request.
S3 bucket has versioning enabled which is requirement for Concourse S3 resource. 
First pipeline is using Terraform to create VPC, networking, security groups and S3 bucket to save state and is triggered by a push to paas-cf repository. 
Second pipeline destroys all objects created by first pipeline and has to be executed manually.

Additionally this PR contains:
- Crown license
- ```.gitignore```
- build status in README file
- Travis configuration
- helper scripts to set up Concourse pipelines

You can find more details in commit messages.

# How to test
Use your Vagrant box to start concourse server.
Execute ```BRANCH=109030954-initial-vpc concourse/scripts/deploy.sh your_env_name```
Go to concourse to check whether pipelines are correctly created.
Push a commit to ```109030954-initial-vpc``` and check if cf-deploy was started and finished successfully. Please make sure you don't merge your test commits.
Execute ```BRANCH=109030954-initial-vpc concourse/scripts/destroy.sh your_env_name```
Watch ```fly``` output to check if terraform destroys VPC.

Please note that due to a bug destroy S3 bucket action will fail. We have created a Terraform [PR](https://github.com/hashicorp/terraform/pull/4168) to fix this.

# Who can test
Anyone but @combor, @mtekel , @actionjack , @jimconner
